widget/src/penrose/monoidal.sty

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright (c) 2024 Yuma Mizuno. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuma Mizuno
+*/
+
+const {
+    -- Minimum size of a row/column in the object grid.
+    gridMinSize = 50
+    gridPadHorizontal = 50
+    gridPadVertical = 50
+    curveOffset = 20
+    labelOffsetFrac = 0.3
+}
+
+layout = [Grid, Arrows, EnsureOnCanvas]
+
+forall Mor2 f {
+    float f.centerX = ? in [Grid, EnsureOnCanvas]
+    float f.centerY = ? in [Grid, EnsureOnCanvas]
+    vec2 f.center = (f.centerX, f.centerY)
+    -- The `textBoxCenter` is where the text box is. It may differ from `center`
+    -- when the label is offset towards one side.
+    vec2 f.textBoxCenter = f.center
+
+    shape f.textBox = Rectangle {
+        center: f.textBoxCenter
+        cornerRadius: 5
+        -- Default size, will be overwritten dynamically.
+        width: 50
+        height: 30
+        fillColor: theme.tooltipBackground
+        strokeColor: theme.tooltipBorder
+    }
+
+    f.left = f.center[0] - f.textBox.width/2.
+    f.right = f.center[0] + f.textBox.width/2.
+    f.top = f.center[1] + f.textBox.height/2.
+    f.bot = f.center[1] - f.textBox.height/2.
+}
+
+forall Id f {
+  f.textBox.strokeWidth = 0.5
+}
+
+forall Core f {
+  f.textBox.strokeWidth = 1.5
+}
```

Comment:
I could barely tell the difference between these; changing cornerRadius or remove the background color might make them easier to differentiate, assuming that's actually desirable.